addons/mass_mailing_sale/models/mailing_mailing.py

```diff
@@ -23,8 +23,16 @@ def _compute_sale_invoiced_amount(self):
         for mass_mailing in self:
             if self.user_has_groups('sales_team.group_sale_salesman') and self.user_has_groups('account.group_account_invoice'):
                 domain = self._get_sale_utm_domain() + [('state', 'not in', ['draft', 'cancel'])]
-                moves = self.env['account.move'].search_read(domain, ['amount_untaxed'])
-                mass_mailing.sale_invoiced_amount = sum(i['amount_untaxed'] for i in moves)
+                moves = self.env['account.move'].search(domain)
```

Comment:
I bug a bit into this and found that the "_convert" method calls "res.currency#_get_rates", which in turns uses both a flush() and a hand written SQL query to retrieve conversion rates.

I'm afraid that this will perform very poorly if you have 5000 moves to convert.
(Because it will imply 5000 flushes and 5000 queries to retrieve rates).

I thought we could group moves by currency and only call the convert method once for every currency but as we use that date of the move it will not work.
Using a static "now()" as the date would make it work but then it will be a bit inaccurate.
Might be a good enough trade-off though, maybe check with the PO?

But in it's current state, I don't think it will scale.